arangod/Cluster/ClusterInfo.cpp

```diff
@@ -3038,7 +3038,7 @@ Result ClusterInfo::ensureIndexCoordinatorInner(  // create index
                   "rolling back index creation.");
             }
 
-            return Result(tmpRes);
+            return Result(tmpRes, *errMsg);
```

Comment:
A [few lines above](https://github.com/arangodb/arangodb/pull/9454/files#diff-ab0fffbe5ac5ed0a2189cc76962b080dR3002) there is a comment saying
> // The mutex in the condition variable protects the access to
> // *errMsg:

I think we should use the CONDITION_LOCKER here too.